### PR TITLE
HTML output share

### PR DESCRIPTION
**Pull request by LeoniePeters**

### CODE REVIEW

**Review by alansutherland — APPROVED** [review]

**Review by lbiedinger — CHANGES_REQUESTED** [review]

File: components/sharing/SocialShareModal.vue [hunk 1]
```diff
@@ -36,6 +68,45 @@
       mediaUrl: {
         type: String,
         default: null
+      },
+      identifier: {
+        type: String,
+        default: null
+      },
+      onItemPage: {
```
Comment (lbiedinger) [inline components/sharing/SocialShareModal.vue:76]: Couldn't this be derived from whether or not the identifier is present?
```
methods: {
  onItemPage() {
    return !this.identifer
  }
  ...
}
```

File: components/sharing/SocialShareModal.vue [hunk 2]
```diff
@@ -5,16 +5,46 @@
     hide-header-close
     hide-footer
     data-qa="share modal"
+    v-on="onItemPage ?
+      { show: requestEmbed,
```
Comment (lbiedinger) [inline components/sharing/SocialShareModal.vue:9]: This results in an attempt to make a request every time the share modal is opened. I say attempt because actually the browser appears to cache it after the first attempt.

One way around this would be to check if `this.oEmbedDataHtml` is present, and if so return early in `requestEmbed`

**Review by lbiedinger — APPROVED** [review]

File: components/sharing/SocialShareModal.vue [hunk 3]
```diff
@@ -5,16 +5,46 @@
...
+      :onItemPage ? { hide: resetEmbedCopied }
+        :{}"
   >
     <div class="icon-wrapper">
       <SocialShare
         :media-url="mediaUrl"
         with-text
       />
     </div>
+    <b-form
```
Comment (rwd) [inline components/sharing/SocialShareModal.vue:19]: This should go into its own component, e.g. /components/item/ItemEmbedCode.vue.

SocialShareModal should then have an unnamed [slot](https://vuejs.org/v2/guide/components-slots.html) at this position, which the ItemHero would use to insert ItemEmbedCode, thus:
```
<SocialShareModal
  :media-url="selectedMedia.about"
>
  <ItemEmbedCode
    :identifier="identifier"
  />
</SocialShareModal>
```

File: components/sharing/SocialShareModal.vue [hunk 4]
```diff
@@ -5,16 +5,46 @@
...
+      v-if="oEmbedDataHtml"
+      data-qa="share modal embed"
+      class="mt-3"
+      @submit.stop.prevent="submitForm"
+    >
+      <label> {{ $t('record.copyEmbedLabel') }}</label>
```
Comment (rwd) [inline components/sharing/SocialShareModal.vue:25]: The label element needs to be linked to the textarea element it labels, with the attribute `for="shareEmbed"`.

Also remove the extraneous space after the opening label tag.

File: components/sharing/SocialShareModal.vue [hunk 5]
```diff
@@ -5,16 +5,46 @@
...
+      <b-form-textarea
+        id="shareEmbed"
+        ref="shareEmbed"
+        v-model="oEmbedDataHtml"
+        readonly
+        tabindex="0"
```
Comment (rwd) [inline components/sharing/SocialShareModal.vue:31]: Why is this needed?

File: components/sharing/SocialShareModal.vue [hunk 6]
```diff
@@ -36,6 +68,41 @@
...
+                          `${EUROPEANA_DATA_URL}/item${this.identifier}`)
+          .then((response) => {
+            if (response.data.html) {
+              this.oEmbedDataHtml = response.data.html;
+            }
+          });
+      },
+      copyEmbedCode(event) {
+        if (event.type === 'click' || event.keyCode === 13) {
+          let textarea = this.$refs.shareEmbed;
```
Comment (rwd) [inline components/sharing/SocialShareModal.vue:98]: This doesn't warrant a variable assignment. Just do `this.$refs.shareEmbed.select();`

File: components/sharing/SocialShareModal.vue [hunk 7]
```diff
@@ -36,6 +68,41 @@
...
+        oEmbedForEndpoint(process.env.OEMBED_ENDPOINT || 'https://oembedjs.europeana.eu/',
+                          `${EUROPEANA_DATA_URL}/item${this.identifier}`)
+          .then((response) => {
+            if (response.data.html) {
+              this.oEmbedDataHtml = response.data.html;
+            }
+          });
+      },
+      copyEmbedCode(event) {
+        if (event.type === 'click' || event.keyCode === 13) {
```
Comment (rwd) [inline components/sharing/SocialShareModal.vue:97]: [keyCode is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode), so if possible we should not introduce further uses of it.

Can this instead use [key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key)?

**Review by rwd — CHANGES_REQUESTED** [review]

File: components/sharing/SocialShareModal.vue (hunk 5 above)
Comment (LeoniePeters) [reply components/sharing/SocialShareModal.vue:31]: For accessibility we would want the textarea to be accessible through keyboard, since it's also clickable. But on reviewing, I guess we don't need to set tabindex, since textarea is an interactive element and already focusable. I believe that at some point I was testing and couldn't reach the textarea via keyboard so added the tabindex. Will take another look at this.

File: components/sharing/SocialShareModal.vue (hunk 7 above)
Comment (LeoniePeters) [reply components/sharing/SocialShareModal.vue:97]: Then the embed code will be copied on any key. This is for keyboard users. So if the user tabs through this module, focuses on the textarea with the embed code and hits tab again, the code will be copied. I think that would be annoying/undesired behavior.

Comment (LeoniePeters) [reply components/sharing/SocialShareModal.vue:97]: [Vue key modifiers](https://vuejs.org/v2/guide/events.html#Key-Modifiers) solves this and makes that conditional unnecessary :+1:

File: components/sharing/SocialShareModal.vue (hunk 5 above)
Comment (LeoniePeters) [reply components/sharing/SocialShareModal.vue:31]: Tested keyboard accessibility and it is indeed not needed to use tabindex.

**Review by rwd — APPROVED** [review]
